pkg/apis/contrail/v1alpha1/contrailcni_types.go

```diff
@@ -0,0 +1,223 @@
+
+// ContrailCNIList contains a list of ContrailCNI
+type ContrailCNIList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ContrailCNI `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&ContrailCNI{}, &ContrailCNIList{})
+}
+
+// CreateConfigMap creates configMap referenced to contrailCNI
+func (c *ContrailCNI) CreateConfigMap(configMapName string,
+	client client.Client,
+	scheme *runtime.Scheme,
+	request reconcile.Request) (*corev1.ConfigMap, error) {
+	return CreateConfigMap(configMapName,
+		client,
+		scheme,
+		request,
+		"contrailcni",
+		c)
+}
+
+// PrepareDaemonSet prepares the intended podList.
+func (c *ContrailCNI) PrepareDaemonSet(ds *appsv1.DaemonSet,
+	commonConfiguration *PodConfiguration,
+	request reconcile.Request,
+	scheme *runtime.Scheme,
+	client client.Client) error {
+	instanceType := "contrailcni"
+	SetDSCommonConfiguration(ds, commonConfiguration)
+	ds.SetName(request.Name + "-" + instanceType + "-daemonset")
+	ds.SetNamespace(request.Namespace)
+	ds.SetLabels(map[string]string{"contrail_manager": instanceType,
+		instanceType: request.Name})
+	ds.Spec.Selector.MatchLabels = map[string]string{"contrail_manager": instanceType,
+		instanceType: request.Name}
+	ds.Spec.Template.SetLabels(map[string]string{"contrail_manager": instanceType,
+		instanceType: request.Name})
+	err := controllerutil.SetControllerReference(c, ds, scheme)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// PodIPListAndIPMapFromInstance gets a list with POD IPs and a map of POD names and IPs.
+func (c *ContrailCNI) PodIPListAndIPMapFromInstance(instanceType string, request reconcile.Request, reconcileClient client.Client, getPhysicalInterface bool, getPhysicalInterfaceMac bool, getPrefixLength bool, getGateway bool) (*corev1.PodList, map[string]string, error) {
+	return PodIPListAndIPMapFromInstance(instanceType, &c.Spec.CommonConfiguration, request, reconcileClient, false, true, getPhysicalInterface, getPhysicalInterfaceMac, getPrefixLength, getGateway)
+}
+
+// SetPodsToReady sets Kubemanager PODs to ready.
+func (c *ContrailCNI) SetPodsToReady(podIPList *corev1.PodList, client client.Client) error {
+	return SetPodsToReady(podIPList, client)
+}
+
+// InstanceConfiguration configure values of instance config maps
+func (c *ContrailCNI) InstanceConfiguration(request reconcile.Request,
+	client client.Client,
+	clusterInfo VrouterClusterInfo,
+	instanceType string) error {
+	instanceConfigMapName := request.Name + "-" + instanceType + "-configuration"
+	configMapInstanceDynamicConfig := &corev1.ConfigMap{}
+	err := client.Get(context.TODO(),
+		types.NamespacedName{Name: instanceConfigMapName, Namespace: request.Namespace},
+		configMapInstanceDynamicConfig)
+	if err != nil {
+		return err
+	}
+
+	clusterName, err := clusterInfo.KubernetesClusterName()
+	if err != nil {
+		return err
+	}
+
+	vrouterConfigInstance := c.ConfigurationParameters()
```

Comment:
Please change the names to cniConfiguration or somethink like that.

Reply:
Done